pkg/controller/tas/node_failure_controller_test.go

```diff
@@ -123,13 +122,15 @@ func TestNodeFailureReconciler(t *testing.T) {
 	baseNode := testingnode.MakeNode(nodeName)
 
 	tests := map[string]struct {
+		enableTASReplaceNodeOnPodTermination bool
+		enableTASFailedNodeReplacement       bool
+
 		initObjs           []client.Object
 		reconcileRequests  []reconcile.Request
 		advanceClock       time.Duration
 		wantUnhealthyNodes []kueue.UnhealthyNode
 		wantRequeue        time.Duration
 		wantEvictedCond    *metav1.Condition
-		featureGates       []featuregate.Feature
```

Comment:
Personally, I find using

```go
featureGates map[features.Feature]bool
```

with a loop like:

```go
for featureGate, enabled := range tc.featureGates {
  features.SetFeatureGateDuringTest(t, featureGate, enabled)
}
```

more flexible than declaring discrete variables or values at the individual test case level.

Reply:
Done